compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -364,14 +365,46 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
           }
         }
 
-        val clDefs = generatedClasses.toList
+        val clDefs = if (generatedStaticForwarderClasses.isEmpty) {
+          /* Fast path, applicable under -Xno-forwarders, as well as when all
+           * the `object`s of a compilation unit have a companion class.
+           */
+          generatedClasses.toList
+        } else {
+          val regularClasses = generatedClasses.toList
+
+          /* #4148 Add generated static forwarder classes, except those that
+           * would collide with regular classes on case insensitive file
+           * systems.
+           */
+
+          /* I could not find any reference anywhere about what locale is used
```

Comment:
What does scalac do to emit the warnings?

Reply:
It uses `.toLowerCase()` without argument, but I think that's very bad because different users could have different warnings.

Reply:
Ah, yes. That is definitely worse than fixing the english locale. Maybe add that in the comment?

Reply:
I've added that in the comment.